### DISCUSSION TITLE
Charon is unable install route in table 220 inside k8s POD.

### DISCUSSION THREAD

**Discussion by shyampujar** [strongswan/strongswan#2386] [Q&A]
I am using strongswan 5.7.2. and swanctl.conf has left-ip configured on the lo device inside the POD. 
when the tunnel gets established, IPSEC SA and Policy are inserted properly but "route addition" for inner ip fails with following error. 

[IKE] scheduling rekeying in 78022s
[IKE] maximum IKE_SA lifetime 86662s
[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/NO_EXT_SEQ
[KNL] received netlink error: Network is unreachable (101)
[KNL] unable to install source route for 192.168.73.10
[IKE] CHILD_SA Child1-bh-x2-ipsec{2} established with SPIs c80a988d_i c4998097_o and TS 192.168.73.10/32 === 0.0.0.0/0
initiate completed successfully

Same strongswan version is able to install routes on normal linux host on Centos 7.9 , Rocky Linux 8.5. 
Could you please comment why does route installation fail inside the k8s POD? 

Regards,
Shyam

**Comment by tobiasbrunner** [comment]
Could be due to `onlink` routes (see the discussion in #1376).

**Reply by shyampujar** [reply]
Thank you for the quick reply. I took sometime to test on different env. 
Here are my findings.
1. Irrespective of /32 or proper subnet, routes do not get inserted by charon inside pod.
2. I have tried on 2 env.  vanila kubernetes and another one is robin platform ( enhancements over k8s ). Both have same results inside the POD. 
3. If i insert manually the route without "via" then route gets inserted and i can ping the remote inner ip as well. 

Is "via" route addition explicitly  done by charon or kernel adds it by default. ?
I could not find any "onlink" settings on the interface as mentioned in the link #1376 

It would be good, if you could comment as "what setting/control inside networking namespace , makes remote inner ip routes insertion happens using via ".
  

Regards,
Shyam

**Reply by shyampujar** [reply]
Hi Tobias, 
I have checked strongswan code. charon does explicit addition of route. However i could not figure out as in one setup
charon add "via" route and in another setup "without via". Both setups are in different customer deployment. 
configuration swanctl.conf remains same in both deployment. Does remote "swanctl.conf/ipsec configuration" matters in adding routes by charon?

Regards,
Shyam
